src/main/java/uk/gov/hmcts/reform/slc/services/servicebus/IReceiverProvider.java

```diff
@@ -0,0 +1,9 @@
+package uk.gov.hmcts.reform.slc.services.servicebus;
+
+import com.microsoft.azure.servicebus.IMessageReceiver;
+
+import java.util.function.Supplier;
+
+@FunctionalInterface
+public interface IReceiverProvider extends Supplier<IMessageReceiver> {
```

Comment:
Can't see the value of it yet... Why to implement `IReceiverProvider` when `Supplier<IMessageReceiver>` provides the same API?

Reply:
so that I can declare a `Component`

Reply:
Would it be much difference if class implemented Supplier, like:

```java
public class ReceiverProvider implements Supplier<IMessageReceiver> {
```
?

Reply:
looks like it works. removing interface.